### PR TITLE
FIX: Validate email_accent_bg_color color

### PR DESCRIPTION
**Pull request by nbianca**
Using an invalid value was allowed. This commit tries to automatically
fix the color by adding missing # symbol or will show an error to the
user if it is not possible and it is not a CSS color either.

### CODE REVIEW

File: app/assets/javascripts/admin/addon/components/color-input.js [hunk 1]
```diff
@@ -22,17 +22,31 @@ export default Component.extend({
     return this.onlyHex ? 6 : null;
   }),
 
+  normalize(color) {
+    if (/^#?([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$/.test(color)) {
```
Comment (ZogStriP) [inline app/assets/javascripts/admin/addon/components/color-input.js:26]: I _think_ you can reduce the regexp to this

```suggestion
    if (/^#?([0-9a-f]{3}){1,2}$/i.test(color)) {
```

File: lib/validators/css_color_validator.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    skyblue slateblue slategray slategrey snow springgreen steelblue tan teal
+    thistle tomato turquoise violet wheat white whitesmoke yellow yellowgreen
+  }
+
+  def initialize(opts = {})
+    @opts = opts
+  end
+
+  def valid_value?(val)
+    val =~ /^#([0-9a-fA-F]{3}|[0-9a-fA-F]{6})$/ || COLORS.include?(val&.downcase)
```
Comment (ZogStriP) [inline lib/validators/css_color_validator.rb:34]: Ditto, you _should_ be able to reduce the regepx to this (ruby has the nice `\h` shorthand for hexadecimals

```suggestion
    val =~ /^#(\h{3}){1,2}$/i || COLORS.include?(val&.downcase)
```

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -2327,6 +2327,7 @@ en:
     create_revision_on_bulk_topic_moves: "Create revision for first posts when topics are moved into a new category in bulk."
 
     errors:
+      invalid_css_color: "Invalid color."
```
Comment (ZogStriP) [inline config/locales/server.en.yml:2330]: Should we be more descriptive / helpful here? Maybe something like

> Invalid color. Either hexadecimal value or a color name.

File: spec/components/validators/css_color_validator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe CssColorValidator do
+  subject { described_class.new }
+
+  it "validates hex colors" do
+    expect(subject.valid_value?('#0')).to be_falsey
```
Comment (ZogStriP) [inline spec/components/validators/css_color_validator_spec.rb:9]: Any reasons why you're not using `eq(false)` or `be false`?

We're rather be explicit about when we expect a method to return a boolean.

File: config/locales/server.en.yml (hunk 3 above)
Comment (nbianca) [reply config/locales/server.en.yml:2330]: The site setting description says:

> The accent color to be used as the background of some elements in HTML emails. Enter a color name ('red') or hex value ('#FF0000').

I noticed all the error messages are short and cut to the point. Do you still want me to make the error more descriptive?

Comment (ZogStriP) [reply config/locales/server.en.yml:2330]: Yes, we should always strive for better error messages.

**Review by ZogStriP — APPROVED** [review]

**Pull request merged**